Dumping Twig_Environment broke


Symfony version(s) affected: 4.3 aka master
Description
Consider
final class HomeController
{
    public function __invoke(\Twig_Environment $twig): Response
    {
        dd($twig);
    }
}
In 4.2 the service is correctly dumped to screen, since 4.3 i get:
Function name must be a string

  at var/symfony-src/src/Symfony/Component/VarDumper/Cloner/AbstractCloner.php:319
  at Symfony\Component\VarDumper\Cloner\AbstractCloner->castObject(object(Stub), true)
     (var/symfony-src/src/Symfony/Component/VarDumper/Cloner/VarCloner.php:182)
  at Symfony\Component\VarDumper\Cloner\VarCloner->doClone(object(Twig_Environment))
     (var/symfony-src/src/Symfony/Component/VarDumper/Cloner/AbstractCloner.php:251)
  at Symfony\Component\VarDumper\Cloner\AbstractCloner->cloneVar(object(Twig_Environment))
     (var/symfony-src/src/Symfony/Bundle/DebugBundle/DebugBundle.php:39)
  at Symfony\Bundle\DebugBundle\DebugBundle->Symfony\Bundle\DebugBundle\{closure}(object(Twig_Environment))
     (var/symfony-src/src/Symfony/Bundle/DebugBundle/DebugBundle.php:42)
  at Symfony\Bundle\DebugBundle\DebugBundle->Symfony\Bundle\DebugBundle\{closure}(object(Twig_Environment))
     (var/symfony-src/src/Symfony/Component/VarDumper/VarDumper.php:46)
  at Symfony\Component\VarDumper\VarDumper::dump(object(Twig_Environment))
     (var/symfony-src/src/Symfony/Component/VarDumper/Resources/functions/dump.php:38)
  at dd(object(Twig_Environment))
     (src/Controller/HomeController.php:23)
  at App\Controller\HomeController->__invoke(object(Twig_Environment))
     (var/symfony-src/src/Symfony/Component/HttpKernel/HttpKernel.php:150)
  at Symfony\Component\HttpKernel\HttpKernel->handleRaw(object(Request), 1)
     (var/symfony-src/src/Symfony/Component/HttpKernel/HttpKernel.php:67)
  at Symfony\Component\HttpKernel\HttpKernel->handle(object(Request), 1, true)
     (var/symfony-src/src/Symfony/Component/HttpKernel/Kernel.php:198)
  at Symfony\Component\HttpKernel\Kernel->handle(object(Request))
     (public/index.php:27)

The invalid callback at this point is:
object(Symfony\Component\VarDumper\Cloner\Stub)#1263 (8) {
  ["type"]=>
  int(1)
  ["class"]=>
  string(0) ""
  ["value"]=>
  object(Symfony\Component\VarDumper\Cloner\Stub)#1252 (8) {
    ["type"]=>
    int(4)
    ["class"]=>
    string(17) "Closure(...$args)"
    ["value"]=>
    NULL
    ["cut"]=>
    int(0)
    ["handle"]=>
    int(512)
    ["refCount"]=>
    int(0)
    ["position"]=>
    int(445)
    ["attr"]=>
    array(2) {
      ["file"]=>
      string(78) "/app/var/symfony-src/src/Symfony/Component/VarDumper/Cloner/AbstractCloner.php"
      ["line"]=>
      int(186)
    }
  }
  ["cut"]=>
  int(0)
  ["handle"]=>
  int(13)
  ["refCount"]=>
  int(0)
  ["position"]=>
  int(0)
  ["attr"]=>
  array(0) {
  }
}

cc @nicolas-grekas
